modules/nf-core/cellbender/merge/templates/merge.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+import platform
+import cellbender
+import anndata as ad
+from cellbender.remove_background.downstream import load_anndata_from_input_and_output
+
+def format_yaml_like(data: dict, indent: int = 0) -> str:
+    """Formats a dictionary to a YAML-like string.
+
+    Args:
+        data (dict): The dictionary to format.
+        indent (int): The current indentation level.
+
+    Returns:
+        str: A string formatted as YAML.
+    """
+    yaml_str = ""
+    for key, value in data.items():
+        spaces = "  " * indent
+        if isinstance(value, dict):
+            yaml_str += f"{spaces}{key}:\\n{format_yaml_like(value, indent + 1)}"
+        else:
+            yaml_str += f"{spaces}{key}: {value}\\n"
+    return yaml_str
+
+
+adata = ad.read_h5ad("${filtered}")
+
+adata_cellbender = load_anndata_from_input_and_output("${unfiltered}", "${cellbender_h5}",
+                                            analyzed_barcodes_only=False)
+
+adata_cellbender = adata_cellbender[adata.obs_names]
+adata.X = adata_cellbender.layers["cellbender"]
```

Comment:
Maybe better to store cellbender in a layer rather than X? Otherwise the original expression values get lost

Reply:
What do you think about this solution:

- Input layer is specified as `task.ext.input_layer`, defaulting to `X`
- Output layer specified as `tesk.ext.output_layer`, defaulting to `cellbender` (or maybe something more general, e.g. `denoised` or `ambient_removed`)

With a little python logic to treat `X` differently

Reply:
That would be the most generic solution... but I think you mean `output_layer` defaults to X? 

Could you elaborate why this process is needed in the first place? It doesn't seem to do a lot and requires reading/writing the anndata object from disk which is inefficient. 

I'm also not sure under which circumstances you'd like to have cellbender counts in `.X` rather than a layer.